handler.go

```diff
@@ -317,8 +317,16 @@ func receiveStreamMessage(ctx context.Context, stream *yarpctransport.ClientStre
 }
 
 // closeSendStream closes the stream from the client side while
-// stream can continue to receive messages from server
-func closeSendStream(ctx context.Context, stream *yarpctransport.ClientStream) error {
+// stream can continue to receive messages from server. If non-zero delay is
+// passed, stream is closed after the delay.
+func closeSendStream(ctx context.Context, stream *yarpctransport.ClientStream, delayCloseSendStream time.Duration) error {
+	if delayCloseSendStream != 0 {
+		select {
+		case <-time.After(delayCloseSendStream):
+		case <-ctx.Done():
```

Comment:
can we cover this case in a test?

Reply:
Added a unit test, also covered in an integration test.